AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -88,122 +93,205 @@ def __init__(self):
+        answer = htmlResponse[self.html_ids['Ranks']].split("\n")
+        if answer[0] != "Add named ranks, rename old/autogenerated ranks, and set default rank." or answer[1] != "The final and penultimate ranks are reserved for Umpire and Chief of Police":
+            return [Label(f"[POLICE] Rename failed - Instructions tampered with")]
+        answer.pop(0)
+        answer.pop(0)  # To remove the instruction lines. Not sure if they are even necessary, but I think they help
+        if len(answer) < 3:
+            # TODO check how many ranks minimum are required from event data.
+            return [Label(f"[POLICE] Rename failed - Must have enough ranks")]
 
-    def on_event_request_create(self) -> List[HTMLComponent]:  # TODO Make a dependent selector, ideally with police filtering
-        return [
-            Dependency(
-                dependentOn="CorePlugin_assassin_pseudonym",
-                htmlComponents=[
-                    AssassinDependentTextEntry(
-                        pseudonym_list_identifier="CorePlugin_assassin_pseudonym",
-                        identifier=self.html_ids["Rank"],
-                        title="Change police rank? Must be a number from 0-9",
-                        default={}
-                    )
-                ]
-            )
-        ]
+        default_counter = 0
+        for idx, ele in enumerate(answer):
+            if ele[-8:] == "=default":
+                default_counter += 1
+                default = idx
+                answer[idx] = ele[:-8]
+        if default_counter == 0:
+            return [Label(f"[POLICE] Rename failed - No default given")]
+        elif default_counter > 1:
+            return [Label(f"[POLICE] Rename failed - Too many defaults given")]
+        elif default > (len(answer)-3):
+            return [Label(f"[POLICE] Rename failed - Umpire/CoP can't be default")]
+
+        self.gsdb_set("Ranks", answer)
+        self.gsdb_set("Default Rank", default)
+        return [Label(f"[POLICE] Success!")]
+
+    def answer_toggle_auto(self, _):
+        self.gsdb_set("Auto Rank", not self.gsdb_get('Auto Rank'))
+        return [Label(f"[POLICE] Toggled Auto Ranking from {not self.gsdb_get('Auto Rank')} to {self.gsdb_get('Auto Rank')}")]
+
+    def answer_toggle_manual(self, _):
+        self.gsdb_set("Manual Rank", not self.gsdb_get('Manual Rank'))
+        return [Label(f"[POLICE] Toggled Manual Ranking from {not self.gsdb_get('Manual Rank')} to {self.gsdb_get('Manual Rank')}")]
+
+    def answer_toggle_police_kills_rankup(self, _):
+        self.gsdb_set("Police Kills Rankup", not self.gsdb_get('Police Kills Rankup'))
+        return [Label(f"[POLICE] Toggled Auto Ranking on Police Kills from {not self.gsdb_get('Police Kills Rankup')} to {self.gsdb_get('Police Kills Rankup')}")]
+
+    def on_event_request_create(self) -> List[HTMLComponent]:
+        # TODO Make a selector with police filtering
+        if self.gsdb_get("Manual Rank"):
```

Comment:
nit: you don't need an if/else when both paths are `return`s - you can drop the `else` as it's redundant. In this case, I'd negate the `if` condition, move `return []` into that branch, and dedent the dependency return.

Applies elsewhere as well.